src/sagas/task.saga.test.js

```diff
@@ -0,0 +1,382 @@
+      saga.next();
+      saga.next(projectReact);
+
+      expect(saga.next(projectPath).value).toEqual(
+        call(
+          [childProcess, childProcess.spawn],
+          PACKAGE_MANAGER_CMD,
+          ['run', name, '--coverage'],
+          {
+            cwd: projectPath,
+            env: getBaseProjectEnvironment(projectPath),
+          }
+        )
+      );
+    });
+
+    it('should run without additionalArgs for gatsby', () => {
+      const name = 'test';
+      const saga = taskRun({ task: { name, projectId: '' } });
+      saga.next();
+      saga.next(projectGatsby);
+
+      expect(saga.next(projectPath).value).toEqual(
+        call(
+          [childProcess, childProcess.spawn],
+          PACKAGE_MANAGER_CMD,
+          ['run', name],
+          {
+            cwd: projectPath,
+            env: getBaseProjectEnvironment(projectPath),
+          }
+        )
+      );
+    });
+
+    it('should run without additional args for create-react-app non-test tasks', () => {
+      const name = 'not-test';
+      const saga = taskRun({ task: { name, projectId: '' } });
+      saga.next();
+      saga.next(projectReact);
+
+      expect(saga.next(projectPath).value).toEqual(
+        call(
+          [childProcess, childProcess.spawn],
+          PACKAGE_MANAGER_CMD,
+          ['run', name],
+          {
+            cwd: projectPath,
+            env: getBaseProjectEnvironment(projectPath),
+          }
+        )
+      );
+    });
+
+    it('should attach task metadata and notify renderer', () => {
+      saga.next(projectReact);
+      saga.next(projectPath);
+
+      expect(saga.next(mockProcess).value).toEqual(
+        put(attachTaskMetadata(task, processId))
+      );
+      expect(saga.next().value).toEqual(
+        call([ipcRenderer, ipcRenderer.send], 'addProcessId', processId)
+      );
+    });
+
+    it('should display logs from stderr', () => {
+      const clone = saga.clone();
+      const text = "Oops, something went wrong but it's probably not fatal";
+
+      // `take` a log message
+      clone.next();
+
+      expect(clone.next({ channel: 'stdout', text }).value).toEqual(
+        put(receiveDataFromTaskExecution(task, text))
+      );
+    });
+
+    it('should complete on exit', () => {
+      const timestamp = new Date();
+
+      // `take` a log message
+      saga.next();
+
+      expect(
+        saga.next({ channel: 'exit', timestamp, wasSuccessful: true }).value
+      ).toEqual(call(displayTaskComplete, task));
+
+      expect(saga.next().value).toEqual(
+        put(completeTask(task, timestamp, true))
+      );
+    });
+
+    it('should reload dependencies after eject', () => {
+      // `loadDependencyInfoFromDisk` is a thunk, thus two copies
+      // constructed with identical arguments will return unique
+      // copies of the same anonymous function. As such, their JSON
+      // stringified representations are used for testing equality.
```

Comment:
I think `JSON.stringify` is stripping out the functions altogether; when I log it, I get `"{\"@@redux-saga/IO\":true,\"PUT\":{\"channel\":null}}"`. No mention of the `action` created.

I think that's OK, though; ideally I want to get rid of redux-thunk altogether, and move it to a saga, so this is a temporary problem :D

Reply:
Yep, `JSON.stringify` skips keys with function values.

Reply:
Yeah, this is one spot where redux-saga and redux-thunk don't play well together. I'll add a TODO to update these once redux-thunk is out of the picture.

Reply:
Happily there's only 1 thunk left, and it should be gone soon!